src/Invictus.Testing.LogicApps/LogicAppAuthentication.cs

```diff
@@ -31,18 +31,24 @@ private LogicAppAuthentication(Func<Task<LogicManagementClient>> authenticateAsy
         /// <param name="clientId">The ID of the client or application that has access to the logic apps running on Azure.</param>
         /// <param name="clientSecretKey">The secret of the client or application that has access to the logic apps running on Azure.</param>
         /// <param name="secretProvider">The provider to get the client secret; using the <paramref name="clientSecretKey"/>.</param>
-        public static LogicAppAuthentication UsingServicePrincipal(string tenantId, string subscriptionId, string clientId, string clientSecretKey, ISecretProvider secretProvider)
+        /// <param name="cloud">The Azure cloud environment to use during authenticating and interacting with the Azure Logic Apps resources.</param>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="tenantId"/>, <paramref name="subscriptionId"/>, <paramref name="clientId"/>, or <paramref name="clientSecretKey"/> is blank.</exception>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="secretProvider"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the <paramref name="cloud"/> is outside the bounds of the enumeration.</exception>
+        public static LogicAppAuthentication UsingServicePrincipal(string tenantId, string subscriptionId, string clientId, string clientSecretKey, ISecretProvider secretProvider, AzureCloud cloud = AzureCloud.Global)
```

Comment:
See comment below
```suggestion
        public static LogicAppAuthentication UsingServicePrincipal(string tenantId, string subscriptionId, string clientId, string clientSecretName, ISecretProvider secretProvider, AzureCloud cloud = AzureCloud.Global)
```

Reply:
A little pirates of the Carabbean flashback (up means down, below means up 😉 ).
![](https://myliveactiondisneyproject.files.wordpress.com/2014/07/potc3upsidedown.jpg?w=529&h=216)